lib/arrays.md

```diff
@@ -5,44 +5,110 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+Arrays are objects that store a list of values. Each value in the list is an element of the array, starting at an index of 0. 
+
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+Addresses, emails, and other large lists of data
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+You can retrieve array elements by using their index or negative index. 
+1.) To retrieve the 6th element, you could do:
+    arr = [1,2,3,4,5,6,7,8]
+        puts arr[5]             #=> 6
+        
+2.) If you want to retrieve the 8th element, you can do arr[7]. If it's the last element in an array, it might be faster to do something like arr[-1]
+
+3.) If you try to return the value of an element that doesn't exist, it will return nil. If you are trying to use 
+floating values like 0.99998 when searching by index, it will disregard the floating numbers and only pay attention to the integers 
+(i.e the numbers to the left of the decimal).  For example:
+ arr = [1,2,3,4,5,6,7,8]
+        puts arr[10]    #=> nil
+        
+ arr = [1,2,3,4,5,6,7,8]
+        puts arr[0.9998]    #=> 1 
+        
+
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+It is possible to find the negative index in an array when working in Ruby,
+as the index will wrap the array to compensate imaginary numbers (i.e. a negative number, but not a postive number)
+up to the length of the string.
+EXAMPLE:
+arr = [1,2,3,4,5,6,7,8]
+puts arr[-6]        #=> 3
+
+The index of the example is 7. The negative index is -8
+
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+Use the .each method with some sort of iteration. FOR EXAMPLE:
+name=["Laura", "Fiona", "Tori"]
+name.each do | i |
+    i.downcase!
+    puts i
+end                     #=> ["laura", "fiona", "tori"]
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+You can use the push or upshift method.
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+name=["Laura", "Fiona", "Tori"]
+name[1].replace("Florence")
```

Comment:
It's true that `name[1].replace("Florence")` does replace `"Fiona"` with `"Florence"`, and I'll always encourage y'all to find and appreciate the various ways to achieve a goal. So nice job :)

I'd also like to point out that `name[1] = "Florence"` is the more common way to do this. Just an FYI–no reply or revision on your part necessary.